src/ln/keysinterface.rs

```diff
@@ -0,0 +1,186 @@
+//! WalletInterface is *not* a wallet, only an interface to bridge between
+//! user wallet and ChannelMonitor. If this last one discover on-chain outputs they will
+//! be send with associate data as events::Event::SpendableOutputs to be at the disposal of wallet.
+//!
+//! KeysInterface is no more a wallet, just an entity to get secret from user wallet and derive
+//! appropriate keyring materials to others lightning components, as such node_id, destination_script.
+//!
+
+use bitcoin::blockdata::transaction::OutPoint;
+use bitcoin::blockdata::script::{Script, Builder};
+use bitcoin::blockdata::opcodes;
+use bitcoin::network::constants::Network;
+use bitcoin::util::bip32::{ExtendedPrivKey, ExtendedPubKey, ChildNumber};
+
+use secp256k1::key::{SecretKey, PublicKey};
+use secp256k1::Secp256k1;
+use secp256k1;
+
+use ln::channel::ChannelKeys;
+use util::events;
+use util::logger::Logger;
+
+use std::sync::Arc;
+
+/// A trait to describe a wallet which sould receive data to be able to spend onchain outputs
+/// fron a lightning channel
+pub trait WalletInterface: Send + Sync {
+	/// Handle an incoming spendable_outputs message from SimpleManyChannelMonitor.
+	/// If SecretKey is None, wallet should use secret key of destination_script previously
+	/// provided to ChannelMonitor.
+	/// If SecretKey is Some, wallet should use it with basepoint_secret for local_delayed following
+	/// BOTL 3 derivation scheme.
+	fn handle_spendable_output(&self, event: events::Event);
```

Comment:
This seems out of date? Looks like it should be removed?

Reply:
Ah yeah sorry, corrected indicating CustomOutputScriptDescriptor doc to know how to spend